projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -1258,7 +1262,13 @@ void CN0411_cmd_cell_const(uint8_t *args, struct cn0411_device *cn0411_dev)
 		cn0411_dev->cell_const = CELL_CONST_HIGH;
 		printf("Cell Constant set to high.\n");
 	} else {
-		printf("Incorrect input value!\n");
+		input_val = atof(arg);
+		if(input_val > 0) {
+			cn0411_dev->cell_const = input_val;
+			printf("Cell Constant set to %.2f.\n", input_val);
+		} else {
```

Comment:
curios: this only allows values greater than zero, which means zero or negative values are treated as invalid;
is this correct/indented ?  or are those values valid as well ?

Reply:
This statement is correct. There are aren't any negative values for cell constants.